### PR TITLE
SNOW-1359009 Change classpaths in pandas general APIs to be `modin.pandas.<>`

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1359009

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Change classpaths for pandas general APIs to be `modin.pandas.<>` instead of `snowflake.snowpark.modin.pandas.[general].<>`.

### CODE REVIEW

**Review by sfc-gh-azhan — APPROVED** [review]

File: docs/source/_templates/autosummary/base.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% if "modin.pandas" in fullname %}
     {% set fullname = "modin.pandas." + objname %}
 {% endif %}
-{% if "modin.pandas.general" in module %}
+{% if "modin.pandas.general" in module or ("modin.pandas" in module and name in ["read_csv", "read_json", "read_parquet", "read_snowflake", "to_snowpark", "to_pandas"])%}
```
Comment (sfc-gh-azhan) [inline docs/source/_templates/autosummary/base.rst:4]: can we just do `if "modin.pandas.io" ...`

Comment (sfc-gh-vbudati) [reply docs/source/_templates/autosummary/base.rst:4]: No, unfortunately that does not work - the input/output functions are not in `modin.pandas.io` module, just `modin.pandas`. Replacing with `modin.pandas` raises a lot of warnings (100+) so it's probably better to have this bad code here haha.

Comment (sfc-gh-azhan) [reply docs/source/_templates/autosummary/base.rst:4]: this will be hard to maintain. Can you add a jira to track this?

Comment (sfc-gh-vbudati) [reply docs/source/_templates/autosummary/base.rst:4]: Yes - here's the JIRA https://snowflakecomputing.atlassian.net/browse/SNOW-1439522

**Pull request merged**